code/game/machinery/telecomms/computers/telemonitor.dm

```diff
@@ -1,126 +1,82 @@
-			dat += "Linked Entities: <ol>"
-			for(var/obj/machinery/telecomms/T in SelectedMachine.links)
-				if(!T.hide)
-					dat += "<li><a href='?src=[REF(src)];viewmachine=[T.id]'>[REF(T.id)] [T.name]</a> ([T.id])</li>"
-			dat += "</ol>"
-
-
-
-	user << browse(dat, "window=comm_monitor;size=575x400")
-	onclose(user, "server_control")
-
-	temp = ""
-	return
-
-
-/obj/machinery/computer/telecomms/monitor/Topic(href, href_list)
-	if(..())
-		return
-
-
-	add_fingerprint(usr)
-	usr.set_machine(src)
-
-	if(href_list["viewmachine"])
-		screen = 1
-		for(var/obj/machinery/telecomms/T in machinelist)
-			if(T.id == href_list["viewmachine"])
-				SelectedMachine = T
-				break
-
-	if(href_list["operation"])
-		switch(href_list["operation"])
-
-			if("release")
-				machinelist = list()
-				screen = 0
-
-			if("mainmenu")
-				screen = 0
-
-			if("probe")
-				if(machinelist.len > 0)
-					temp = "<font color = #D70B00>- FAILED: CANNOT PROBE WHEN BUFFER FULL -</font color>"
-
-				else
-					for(var/obj/machinery/telecomms/T in urange(25, src))
-						if(T.network == network)
-							machinelist.Add(T)
-
-					if(!machinelist.len)
-						temp = "<font color = #D70B00>- FAILED: UNABLE TO LOCATE NETWORK ENTITIES IN \[[network]\] -</font color>"
-					else
-						temp = "<font color = #336699>- [machinelist.len] ENTITIES LOCATED & BUFFERED -</font color>"
-
-					screen = 0
-
-
-	if(href_list["network"])
-
-		var/newnet = stripped_input(usr, "Which network do you want to view?", "Comm Monitor", network)
-		if(newnet && (get_dist(usr, src) <= 1 || issilicon(usr)))
-			if(length(newnet) > 15)
-				temp = "<font color = #D70B00>- FAILED: NETWORK TAG STRING TOO LENGHTLY -</font color>"
-
-			else
-				network = newnet
-				screen = 0
-				machinelist = list()
-				temp = "<font color = #336699>- NEW NETWORK TAG SET IN ADDRESS \[[network]\] -</font color>"
-
-	updateUsrDialog()
-	return
-
-/obj/machinery/computer/telecomms/monitor/attackby()
-	. = ..()
-	updateUsrDialog()
+    name = "telecommunications monitoring console"
+    icon_screen = "comm_monitor"
+    desc = "Monitors the details of the telecommunications network it's synced with."
+    circuit = /obj/item/circuitboard/computer/comm_monitor
+    network_id = __NETWORK_SERVER // if its connected to the default one we will ignore it
+    var/network = "NULL"		// the network to probe
+    var/list/servers = list()	// the servers in the network
+
+
+/obj/machinery/computer/telecomms/monitor/Initialize(mapload)
+    . = ..()
+    update_network()
+    RegisterSignal(src, COMSIG_COMPONENT_NTNET_RECEIVE, PROC_REF(ntnet_receive))
+
+
+/obj/machinery/computer/telecomms/monitor/ui_interact(mob/user, datum/tgui/ui)
+  ui = SStgui.try_update_ui(user, src, ui)
+  if(!ui)
+    ui = new(user, src, "Telemonitor")
+    ui.set_autoupdate(TRUE)
+    ui.open()
+
+/obj/machinery/computer/telecomms/monitor/ui_act(action, params)
+  if(..())
+    return
```

Comment:
do `. = ..()` first, then do the return as `if(.) return`

Reply:
done